src/GitHub.TeamFoundation.14/Base/TeamExplorerServiceHolder.cs

```diff
@@ -1,52 +1,44 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
-using System.Diagnostics;
-using System.Globalization;
 using System.Linq;
 using System.Threading;
 using GitHub.Extensions;
-using GitHub.Logging;
 using GitHub.Models;
 using GitHub.Services;
 using Microsoft.TeamFoundation.Controls;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.TeamFoundation.Git.Extensibility;
-using Serilog;
 
 namespace GitHub.VisualStudio.Base
 {
     [Export(typeof(ITeamExplorerServiceHolder))]
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class TeamExplorerServiceHolder : ITeamExplorerServiceHolder
     {
-        static readonly ILogger log = LogManager.ForContext<TeamExplorerServiceHolder>();
         readonly Dictionary<object, Action<ILocalRepositoryModel>> activeRepoHandlers = new Dictionary<object, Action<ILocalRepositoryModel>>();
         ILocalRepositoryModel activeRepo;
         bool activeRepoNotified = false;
 
         IServiceProvider serviceProvider;
-        IVSGitExt gitService;
-        IVSUIContext gitUIContext;
-        IVSUIContextFactory uiContextFactory;
+        readonly IVSGitExt gitService;
 
         // ActiveRepositories PropertyChanged event comes in on a non-main thread
         readonly SynchronizationContext syncContext;
 
         /// <summary>
-        /// This class relies on IVSUIContextFactory to get the UIContext object that provides information
-        /// when VS switches repositories. Unfortunately, for some reason MEF fails to create the instance
-        /// when imported from the constructor, so it's imported manually when first accessed via the
-        /// ServiceProvider instance (when mocking, make sure that the ServiceProvider includes this factory)
+        /// This class relies on IVSGitExt that provides information when VS switches repositories.
         /// </summary>
-        /// <param name="gitService"></param>
+        /// <param name="gitService">Used for monitoring the active repository.</param>
         [ImportingConstructor]
         public TeamExplorerServiceHolder(IVSGitExt gitService)
         {
-            this.GitService = gitService;
+            this.gitService = gitService;
             syncContext = SynchronizationContext.Current;
+
+            UpdateActiveRepo();
```

Comment:
This cannot happen here, it will call `ActiveRepositories` which is a potentially blocking call. It needs to be shuffled to a background thread.

Reply:
I've shuffled the initial fetching of `ActiveRepositories` to a thread pool thread so this is now safe to call.